### ISSUE TITLE
TiCDC captures unnecessary changes from TiDB DDL reorganization

### ISSUE THREAD

**Issue by overvenus** [pingcap/tiflow#2434]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

Execute following SQL.

```sql
CREATE TABLE sbtest(
  id INTEGER NOT NULL AUTO_INCREMENT,
  k INTEGER DEFAULT '0' NOT NULL,
  c CHAR(120) DEFAULT '' NOT NULL,
  pad CHAR(60) DEFAULT '' NOT NULL,
  PRIMARY KEY (id)
) /*! ENGINE = Innodb */;

CREATE INDEX k_6 ON sbtest(k);

-- Insert many records
-- INSERT INTO sbtest(k, c, pad) VALUES (...), ... (...);

ALTER TABLE sbtest MODIFY COLUMN k VARCHAR(10);
```

After `MODIFY COLUMN`, CDC sees unexpected row data changes these are caused by DDL reorganization (rewrites all row data). CDC translates them into `DELETE` and `REPLACE INTO`, and executes them on the downstream TiDB/MySQL.

These `DELETE` and `REPLACE INTO` cloud lead to high replication lag.

2. What did you expect to see?

Do not replicate row changes (caused by DDL reorganization) to the downstream, as `MODIFY COLUMN` has been executed on the downstream.

3. What did you see instead?

Still replicates row changes.

4. Versions of the cluster

v5.1.0

**Comment by amyangfei** [comment]
All DDL events are captured by TiCDC because these events share the same key ranges, yet the events in intermediate states will be ignored in TiCDC.
What is the purpose to don't capture these changes, to save network traffic, or reduce the pressure in TiKV?
Besides capturing DDL changes does have many problems, such as schema storage pressure in memory, DDL reentrant problem.

**Comment by overvenus** [comment]
> All DDL events are captured by TiCDC because these events share the same key ranges, yet the events in intermediate states will be ignored in TiCDC.

I'm not talking about DDL itself, but row data changes caused DDL (reorganization).

> What is the purpose to not capture these changes, to save network traffic, or reduce the pressure in TiKV?

These changes (`DELETE` and `REPLACE INTO`) cloud lead to high replication lag.

**Comment by amyangfei** [comment]
> > All DDL events are captured by TiCDC because these events share the same key ranges, yet the events in intermediate states will be ignored in TiCDC.
> 
> I'm not talking about DDL itself, but row data changes caused DDL (reorganization).
> 
> > What is the purpose to not capture these changes, to save network traffic, or reduce the pressure in TiKV?
> 
> These changes (`DELETE` and `REPLACE INTO`) cloud lead to high replication lag.

Is old value enabled, if enabled, changes caused by ddl reorganization should not be captured.

**Comment by overvenus** [comment]
I think old value is irrelevant. The DDL is `MODIFY COLUMN` not `ADD INDEX`.

**Comment by zimulala** [comment]
@overvenus 
This "column type change" operation also backfills the index when reorg, does it also need special treatment in CDC?

**Comment by overvenus** [comment]
TiCDC can workaround index-backfill by enabling old value feature (no need to capture index changes).

**Comment by nongfushanquan** [comment]
This is a new feature which is implemented in the version 7.1
